hdate/htables.py

```diff
@@ -165,131 +181,192 @@ def move_if_not_on_dow(original, replacement, dow_not_orig, dow_replacement):
                  DESC(u"חול המועד פסח", u"חוה\"מ פסח"))),
-    HOLIDAY(1, "pesach_vii", (21, 7), "", [],
+    HOLIDAY(HolidayTypes.YOM_TOV, "pesach_vii", 
+            (21, Months.Nisan), "", [],
             LANG(u"Pesach VII", DESC(u"שביעי פסח", u"ז' פסח"))),
-    HOLIDAY(6, "yom_haatzmaut", ([3, 4, 5], 8), "",
+    HOLIDAY(HolidayTypes.MODERN_HOLIDAY, "yom_haatzmaut", 
+            ([3, 4, 5], Months.Iyyar), "",
             [year_is_after(5708), year_is_before(5764),
              move_if_not_on_dow(5, 4, 4, 3) or
              move_if_not_on_dow(5, 3, 5, 3)],
             LANG(u"Yom HaAtzma'ut", DESC(u"יום העצמאות", u"יום העצמאות"))),
-    HOLIDAY(6, "yom_haatzmaut", ([3, 4, 5, 6], 8), "",
+    HOLIDAY(HolidayTypes.MODERN_HOLIDAY, "yom_haatzmaut", 
+            ([3, 4, 5, 6], Months.Iyyar), "",
             [year_is_after(5763),
              move_if_not_on_dow(5, 4, 4, 3) or
              move_if_not_on_dow(5, 3, 5, 3) or
              move_if_not_on_dow(5, 6, 0, 1)],
             LANG(u"Yom HaAtzma'ut", DESC(u"יום העצמאות", u"יום העצמאות"))),
-    HOLIDAY(7, "lag_bomer", (18, 8), "", [],
+    HOLIDAY(HolidayTypes.MINOR_HOLIDAY, "lag_bomer", 
+            (18, Months.Iyyar), "", [],
             LANG(u"Lag B'Omer", DESC(u"ל\"ג בעומר", u"ל\"ג בעומר"))),
-    HOLIDAY(2, "erev_shavuot", (5, 9), "", [],
+    HOLIDAY(HolidayTypes.EREV_YOM_TOV, "erev_shavuot", 
+            (5, Months.Sivan), "", [],
             LANG(u"Erev Shavuot", DESC(u"ערב שבועות", u"ערב שבועות"))),
-    HOLIDAY(1, "shavuot", (6, 9), "", [],
+    HOLIDAY(HolidayTypes.YOM_TOV, "shavuot", 
+            (6, Months.Sivan), "", [],
             LANG(u"Shavuot", DESC(u"שבועות", u"שבועות"))),
-    HOLIDAY(5, "tzom_tammuz", ([17, 18], 10), "",
+    HOLIDAY(HolidayTypes.FAST_DAY, "tzom_tammuz", 
+            ([17, 18], Months.Tammuz), "",
             [move_if_not_on_dow(17, 18, 5, 6)],
             LANG(u"Tzom Tammuz", DESC(u"צום שבעה עשר בתמוז", u"צום תמוז"))),
-    HOLIDAY(5, "tisha_bav", ([9, 10], 11), "",
+    HOLIDAY(HolidayTypes.FAST_DAY, "tisha_bav", 
+            ([9, 10], Months.Av), "",
             [move_if_not_on_dow(9, 10, 5, 6)],
             LANG(u"Tish'a B'Av", DESC(u"תשעה באב", u"ט' באב"))),
-    HOLIDAY(7, "tu_bav", (15, 11), "", [],
+    HOLIDAY(HolidayTypes.MINOR_HOLIDAY, "tu_bav", 
+            (15, Months.Av), "", [],
             LANG(u"Tu B'Av", DESC(u"ט\"ו באב", u"ט\"ו באב"))),
-    HOLIDAY(8, "yom_hashoah", ([26, 27, 28], 7), "",
+    HOLIDAY(HolidayTypes.HOLOCAUST_MEMORIAL, "yom_hashoah", 
+            ([26, 27, 28], Months.Nisan), "",
             [move_if_not_on_dow(27, 28, 6, 0) or
              move_if_not_on_dow(27, 26, 4, 3),
              year_is_after(5718)],
             LANG(u"Yom HaShoah", DESC(u"יום השואה", u"יום השואה"))),
-    HOLIDAY(8, "yom_hazikaron", ([2, 3, 4], 8), "",
+    HOLIDAY(HolidayTypes.HOLOCAUST_MEMORIAL, "yom_hazikaron", 
+            ([2, 3, 4], Months.Iyyar), "",
             [year_is_after(5708), year_is_before(5764),
```

Comment:
Please rename this,. Yom hazikaron is the memorial day for fallen soldiers. Maybe MEMORIAL_DAY?